src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbol.cs

```diff
@@ -703,6 +704,13 @@ internal override void AfterAddingTypeMembersChecks(ConversionsBase conversions,
                 Debug.Assert(explicitInterfaceSpecifier != null);
                 _explicitInterfaceType.CheckAllConstraints(conversions, new SourceLocation(explicitInterfaceSpecifier.Name), diagnostics);
             }
+
+            if (_refKind == RefKind.RefReadOnly)
```

Comment:
I don't think the general rule applies, because for some symbols like `SourceConstructorMethod` we need the parameters to be computed first before we perform that check, and calling `Parameters` in `AfterAddingTypeMembersChecks` would call `MethodChecks` in turn.
For `SourcePropertySymbol`, it wasn't necessary since it computes parameters without calling `MethodChecks`. Some symbols I believe don't even use this API like `LocalFunctionSymbol`.